### PR TITLE
Suites() coverage

### PR DESCRIPTION
**Pull request by fejta**
Also Artifacts()

### CODE REVIEW

**Comment by fejta** [comment]
Fixed bazel formatting error

File: util/gcs/read_test.go [hunk 1]
```diff
@@ -212,6 +222,378 @@ func TestReadJSON(t *testing.T) {
...
 }
 
+type fakeIterator struct {
+	objects []storage.ObjectAttrs
+	idx     int
+	err     int
+}
+
+func (fi *fakeIterator) Next() (*storage.ObjectAttrs, error) {
+	if fi.idx > len(fi.objects) {
```
Comment (michelle192837) [inline util/gcs/read_test.go:232]: Shouldn't this be >=, not >?

File: util/gcs/read_test.go [hunk 2]
```diff
@@ -212,6 +222,378 @@ func TestReadJSON(t *testing.T) {
...
+		return nil, o.open
+	}
+	return ioutil.NopCloser(&fakeReader{
+		buf:     bytes.NewBufferString(o.data),
+		readErr: o.read,
+	}), nil
+}
+
 // TODO(fejta): TestStarted
+func TestStarted(t *testing.T) {
```
Comment (michelle192837) [inline util/gcs/read_test.go:283]: Do we need the boilerplate for TestStarted? (Or was this in-progress?)

File: util/gcs/read_test.go (hunk 1 above)
Comment (fejta) [reply util/gcs/read_test.go]: Done

File: util/gcs/read_test.go (hunk 2 above)
Comment (fejta) [reply util/gcs/read_test.go:287]: I have a PR out to fill this out, just felt like this PR was already large enough

File: util/gcs/read.go [hunk 3]
```diff
@@ -48,16 +51,26 @@ type Finished struct {
 	Running bool
 }
 
+// An iterator returns the attributes of the listed objects or an iterator.Done error.
+type Iterator interface {
+	Next() (*storage.ObjectAttrs, error)
+}
+
+// An interrogator lists objects and opens them for reading.
+type Interrogator interface {
```
[the remark this answers is not in the document]
Comment (fejta) [reply util/gcs/read.go:60]: Asker hurts my ears but is apparently a word! Can I send a followup PR with that change?

[the remark this answers is not in the document]
Comment (michelle192837) [reply util/gcs/read.go:60]: XD

Yup, sounds good!

**Pull request merged**